internal/ingress/controller/store/store_test.go

```diff
@@ -740,3 +740,76 @@ func TestUpdateSecretIngressMap(t *testing.T) {
 		}
 	})
 }
+
+// Test to avoid regression fixed in https://github.com/kubernetes/ingress-nginx/pull/2361
+func TestListIngressWithoutPath(t *testing.T) {
+	s := newStore(t)
+
+	ing1 := &v1beta1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test",
+			Namespace: "testns",
+		},
+		Spec: v1beta1.IngressSpec{
+			Backend: &v1beta1.IngressBackend{
+				ServiceName: "demo",
+				ServicePort: intstr.FromInt(80),
+			},
+			Rules: []v1beta1.IngressRule{
+				{
+					Host: "foo.bar",
+				},
+			},
+		},
+	}
+	s.listers.Ingress.Add(ing1)
+
+	ing2 := &v1beta1.Ingress{
```

Comment:
Could you call this `ingressToIgnore` or `ingressWithDifferentClass` that way the test case will also serve as a documentation and it will make it easier to figure out that this is the one that we are expecting to be ignored by `ListIngresses` function.

Reply:
done